ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt

```diff
@@ -0,0 +1,175 @@
+package com.braintreepayments.api
+
+import androidx.annotation.VisibleForTesting
+import com.braintreepayments.api.ShopperInsightsAnalytics.GET_RECOMMENDED_PAYMENTS_FAILED
+import com.braintreepayments.api.ShopperInsightsAnalytics.GET_RECOMMENDED_PAYMENTS_STARTED
+import com.braintreepayments.api.ShopperInsightsAnalytics.GET_RECOMMENDED_PAYMENTS_SUCCEEDED
+import com.braintreepayments.api.ShopperInsightsAnalytics.PAYPAL_PRESENTED
+import com.braintreepayments.api.ShopperInsightsAnalytics.PAYPAL_SELECTED
+import com.braintreepayments.api.ShopperInsightsAnalytics.VENMO_PRESENTED
+import com.braintreepayments.api.ShopperInsightsAnalytics.VENMO_SELECTED
+
+/**
+ * Use [ShopperInsightsClient] to optimize your checkout experience
+ * by prioritizing the customer’s preferred payment methods in your UI.
+ * By customizing each customer’s checkout experience,
+ * you can improve conversion, increase sales/repeat buys and boost user retention/loyalty.
+ *
+ * Note: **This feature is in beta. It's public API may change in future releases.**
+ */
+@ExperimentalBetaApi
+class ShopperInsightsClient @VisibleForTesting internal constructor(
+    private val api: ShopperInsightsApi,
+    private val braintreeClient: BraintreeClient
+) {
+    constructor(braintreeClient: BraintreeClient) : this(
+        ShopperInsightsApi(EligiblePaymentsApi(braintreeClient)),
+        braintreeClient,
+    )
+
+    /**
+     * Retrieves recommended payment methods based on the provided shopper insights request.
+     *
+     * @param context Android context
+     * @param request The [ShopperInsightsRequest] containing information about the shopper.
+     * @return A [ShopperInsightsResult] object indicating the recommended payment methods.
+     * Note: This feature is in beta. Its public API may change or be removed in future releases
+     * PayPal recommendation is only available for US, AU, FR, DE, ITA, NED, ESP, Switzerland and
+     * UK merchants. Venmo recommendation is only available for US merchants.
+     */
+    fun getRecommendedPaymentMethods(
+        request: ShopperInsightsRequest,
+        callback: ShopperInsightsCallback
+    ) {
+        braintreeClient.sendAnalyticsEvent(GET_RECOMMENDED_PAYMENTS_STARTED)
+
+        if (request.email == null && request.phone == null) {
+            callbackFailure(
+                callback = callback,
+                error = IllegalArgumentException(
+                    "One of ShopperInsightsRequest.email or ShopperInsightsRequest.phone must be " +
+                            "non-null."
+                )
+            )
+            return
+        }
+
+        // TODO: get correct merchant ID from SDK
+        val merchantId = "MXSJ4F5BADVNS"
```

Comment:
This hardcoded merchant ID needs to get replaced.

Reply:
Hey @agedd - are we dynamically setting merchant ID on iOS?

Reply:
It doesn't look like we have the merchant ID at all on iOS: https://github.com/braintree/braintree_ios/blob/payment-insights-feature/Sources/BraintreeShopperInsights/BTEligiblePaymentsRequest.swift

Reply:
Oh interesting - I think the backend is determining which merchant based on the auth fingerprint. I'll try removing the merchant ID from the request.

Reply:
hi @tdchow! yep we removed it [here](https://github.com/braintree/braintree_ios/pull/1195) since it's now determined by the auth fingerprint.

Reply:
Thanks @agedd! I'll make the same change on Android.